### PR TITLE
feat(components): add a way to manually set text-field in error state (VIV-970)

### PR DESCRIPTION
**Pull request by olaf-k**
There are two ways to go about this:
- a purely cosmetic approach that simply styles the component and displays the message.
- a cleaner way that actually sets the component in error state (caught by `:invalid` selectors, with a proper `validity` prop).

This PR is an attempt at the latter (the former should be quite simple to implement).

Code notes:

- had to "manually" override `validate` in `force-error.ts` because it is added as a mixin *after* the TextField class definition.
- I'm not sure what `userValid` is supposed to represent? though it is needed to refresh the template properly.
- an error message set with `force-error` will supersede any existing error state (conversely, removing it will return the field to its original error state, if any).

### CODE REVIEW

**Review by YonatanKra — CHANGES_REQUESTED** [review]
This PR indeed exposes a missing API for setting a custom error message.

It already exists in the HTML spec: [setCustomValidity](https://developer.mozilla.org/en-US/docs/Web/API/HTMLObjectElement/setCustomValidity)

We should adhere to it IMO.

Since there's no attribute in the spec, it begs the question should we implement one? Diverting from the spec - especially in form elements - might interfere with our components being compatible with the ElementInternals spec which is the future of form association for custom elements.

File: libs/components/src/lib/text-field/text-field.spec.ts [hunk 1]
```diff
@@ -573,4 +573,60 @@ describe('vwc-text-field', () => {
...
+		});
+
+		it('should display the given error message', async function () {
+			element.forceError = forcedErrorMessage;
+			await elementUpdated(element);
+			const errorElement = element.shadowRoot?.querySelector('.error-message');
+			expect(errorElement).toBeDefined();
+		});
+
+		it('should replace/restore the current error state, if any, when set/removed', async function () {
```
Comment (YonatanKra) [inline libs/components/src/lib/text-field/text-field.spec.ts:603]: Can you split this into two `its`, please? One for `set` and one for `remove`?

**Review by YonatanKra — CHANGES_REQUESTED** [review]

**Review by acelletti — CHANGES_REQUESTED** [review]
@olaf-k Can we change `force-error` to `error-message` as it describes what it is that we are passing here.

File: libs/components/src/shared/patterns/force-error.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+		@attr({ attribute: 'force-error' }) forceError?: string;
+		#shouldValidate = true;
+		#prevSuccessText = '';
+
+		constructor(...args: any[]) {
+			super(...args);
+			this._validate = this.validate;
+			this.validate = () => {
+				if (this.#shouldValidate) this._validate();
+			};
```
[the remark this answers is not in the document]
Comment (olaf-k) [reply libs/components/src/shared/patterns/force-error.ts:21]: see the PR notes https://github.com/Vonage/vivid-3/pull/1103#issue-1654086388

**Review by acelletti — APPROVED** [review]

**Pull request merged**